source/data-api.txt

```diff
@@ -53,297 +74,71 @@ The Data API supports two types of endpoints:
-             }
-       
-       .. tab:: JSON
-          :tabid: json
-          
-          .. code-block:: json
-             
-             {
-                "Name": "Mango",
-                "Year": 2022,
-                "Weight": "9823.1297",
-                "Date": "2022-01-12T02:33:23.067Z"
-             }
-
-You define a single default return type for all generated Data API
-endpoints and individually for each custom endpoint. Incoming requests
-can also specify a preferred data format that overrides the default
-using an ``Accept`` header.
-
-The following table compares the JSON and EJSON representations of each
-supported type:
-
-.. list-table::
-   :header-rows: 1
-   :widths: 15 42 42
-   
-   * - Type
-     - EJSON
-     - JSON
-   
-   * - ``Array``
-     
-     - .. code-block:: json
-          :copyable: false
-
-          [ <elements> ]
-     
-     - .. code-block:: json
-          :copyable: false
-
-          [ <elements> ]
-
-   * - ``Binary``
-     
-     - .. code-block:: json
-          :copyable: false
-
-          {
-             "$binary": {
-                "base64": "e67803a39588be8a95731a21e27d7391",
-                "subType": "05"
-             }
-          }
-     
-     - .. code-block:: json
-          :copyable: false
+For each incoming request, the Data API:
 
-          {
-             "Subtype": 5,
-             "Data": "e67803a39588be8a95731a21e27d7391"
-          }
+1. **Authenticates the calling user.** This might involve validating an
+   access token, logging in with header credentials, or directly
+   assigning a specific runtime user based on your configuration.
 
-   * - ``Date``
-     
-     - .. code-block:: json
-          :copyable: false
+2. **Authorizes the request.** This ensures that the user sent a
+   well-formed request and has permission to perform the requested
+   operation based on your endpoint authorization scheme.
 
-          {
-             "$date": {
-                "$numberLong": "1641954803067"
-             }
-          }
-     
-     - .. code-block:: json
-          :copyable: false
+3. **Runs the requested operation.** This might involve reading or
+   writing data in Atlas with a generated endpoint or invoke a custom
+   function that you wrote.
 
-          "2022-01-12T02:33:23.067Z"
+   For requests that read or write data in Atlas, the Data API also
+   enforces the access control rules and document schemas defined in
+   your App. This means that users can only access data they're allowed
+   to read and that requests will fail if they include an invalid write
+   operation.
 
-   * - ``Decimal128``
-     
-     - .. code-block:: json
-          :copyable: false
+4. **Returns an HTTPS response to the caller.** The response includes
+   the result of a generated endpoint operation or any data that you
+   return from a custom endpoint. Requests can choose to receive the
+   response in either :ref:`JSON or EJSON format
+   <data-api-data-formats>`.
```

Comment:
the last sentence reads a little funny to me b/c it seems to assign agency to the request. consider rephrase to assign agency to developer: 

```suggestion
   return from a custom endpoint. In the request, you can choose to receive the
   response in either :ref:`JSON or EJSON format
   <data-api-data-formats>`.
```